pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -167,3 +168,34 @@ func validateFlavor(flavorName string, ci *CloudInfo, req flavorRequirements, fl
 
 	return field.ErrorList{field.Invalid(fldPath, flavor.Name, errString)}
 }
+
+// validateMasterServerGroupPolicy checks that the passed string corresponds to one
+// of the known valid Nova scheduler hints.
+func validateMasterServerGroupPolicy(serverGroupPolicy string, fldPath *field.Path) field.ErrorList {
+	// The field is optional
+	if serverGroupPolicy == "" {
+		return nil
+	}
+
+	for _, knownPolicy := range [...]string{
+		"affinity",
+		"anti-affinity",
+		"soft-affinity",
+		"soft-anti-affinity",
+	} {
+		if serverGroupPolicy == knownPolicy {
+			return nil
+		}
+	}
```

Comment:
Using a set give somewhat better readability of intention.
```suggestion
	knownPolicies := sets.NewString("affinity", "anti-affinity", "soft-affinity", "soft-anti-affinity")
	if knownPolicies.Has(serverGroupPolicy) {
		return nil
	}
```

Reply:
Following your advice I have switched an array for a map, which is the most similar to a set that does not require yet another external dependency.

Reply:
We already have the external dependency.